### PR TITLE
[DPE-3949] Initial commit

### PR DESCRIPTION
**Pull request by welpaolo**
This PR contains the same code already approved in a previous PR.

### CODE REVIEW

**Review by theoctober19th — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+authors = []
+
+[tool.poetry.dependencies]
+python = ">3.8,<4.0"
+ops = ">=2.10.1"
+pyyaml = ">=6.0.1"
+boto3 = ">=1.34.55"
+jsonschema = "^4.19.1"
+lightkube = ">=0.15.2"
+spark8t = { git = "https://github.com/canonical/spark-k8s-toolkit-py.git", branch="dpe-3519"}
```
Comment (theoctober19th) [inline pyproject.toml:38]: I guess we can already pin this to 0.0.7 now.

Comment (welpaolo) [reply pyproject.toml]: Moved to test dependencies

File: metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: spark-integration-hub-k8s
+display-name: Spark Integration Hub
```
Comment (deusebio) [inline metadata.yaml:5]: I believe we use K8s here, elsewhere, see [here](https://github.com/canonical/kafka-k8s-operator/blob/main/metadata.yaml#L4)

File: metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+summary: Kubernetes based charm for the Spark Integration Hub
+description: This is the Spark integration hub charm.
```
Comment (deusebio) [inline metadata.yaml:7]:
```suggestion
description: This is the Spark integration hub charm. It provide centralization for other charmed components and allows simple configuration of Spark jobs and integration with client applications, abstracting low-level configuration option settings.  
```

**Review by deusebio — APPROVED** [review]

**Pull request merged**